code/WorkInProgress/zamu_mail.dm

```diff
@@ -628,6 +756,46 @@ var/global/mail_types_by_job = list(
 		)
 	)
 
+// =========================================================================
+// Various items given out based on player trait choices, to make it more personal
+var/global/mail_types_by_trait = list(
+	"petasusaphilic" = filtered_concrete_typesof(/obj/item/clothing/head, /proc/filter_trait_hats),
+	"pawnstar" = trinket_safelist,
+	"petperson" = filtered_concrete_typesof(/mob/living/critter/small_animal/, GLOBAL_PROC_REF(filter_carrier_pets)),
```

Comment:
Since this is something only people roll with pets, they typically would already have a carrier. It's all goofy anyway and QM sends packages of pets in crates so this isn't too different.